### DISCUSSION TITLE
Fastlane pem not able to login

### DISCUSSION THREAD

**Discussion by MaryArbs** [fastlane/fastlane#21077] [Q&A]
I was hoping to get some help with logging in using fastlane pem. The fastlane document (https://docs.fastlane.tools/actions/pem/) do not speak to login in requirements. We use an api key with all of our apps, however, you can not use an api key with pem. We are trying to get pem to work, logging in with our Apple username, password, and the 2fa requirement. We have set FASTLANE_SESSION and FASTLANE_PASSWORD as global variables within GitLab, but we are not sure how to connect this to our `get_push_certificate` method within our Fastfile. Any help would be appreciated! Thank you :)

**Comment by bassamtarek-st** [comment]
Adding to this, pem depends on the login instead of the apis, this logs you in with a session that expires in a day or so, This defies the concept of the automation

Can anyone get a longer Session Period? or use apis for pem ?
